src/clients/detector.rs

```diff
@@ -122,3 +128,37 @@ impl From<ContentAnalysisResponse> for crate::models::TokenClassificationResult
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn test_text_contents() {
+        let mut mock_client = DetectorClient::faux();
+
+        let request = ContentAnalysisRequest {
+            contents: vec!["My e-mail is me@mail.com".to_string()],
+        };
+        let model_id = "pii";
+
+        let expected_response = vec![vec![ContentAnalysisResponse {
+            start: 13,
+            end: 24,
+            text: "me@mail.com".to_string(),
+            detection: "EmailAddress".to_string(),
+            detection_type: "pii".to_string(),
+            score: 0.8,
+            evidences: vec![].into(),
+        }]];
+
+        faux::when!(mock_client.text_contents(model_id, request.clone()))
+            .once()
+            .then_return(Ok(expected_response.clone()));
+
+        assert_eq!(
+            mock_client.text_contents(model_id, request).await.unwrap(),
+            expected_response
+        );
```

Comment:
Does this test make sense?
I created this inspired by [this](https://github.com/foundation-model-stack/fms-guardrails-orchestrator/blob/b3231a54c5d95e1caf9f036353c9965952dde31a/src/clients/tgis.rs#L98) but it doesn't make sense to me as I'm mocking the very test I'm supposed to test.

Reply:
Agreed, @declark1. I did this one kind of "just copying" the test I pasted as reference, but this test doesn't make sense to me either. I'd vote to remove it.

Reply:
Test removed.